samza-api/src/main/java/org/apache/samza/scheduling/Scheduler.java

```diff
@@ -0,0 +1,28 @@
+package org.apache.samza.scheduling;
+
+import org.apache.samza.task.TimerCallback;
+
+
+/**
+ * Provides a way for applications to register some logic to be executed at a future time.
+ */
+public interface Scheduler {
```

Comment:
I'd prefer to just use Timer/TimerRegistry for this. It'll be confusing to explain three separate time-based-callback concepts - Scheduler, Timer, Window.